PlutoHelperAgent/PreferencesWindowController.m

```diff
@@ -0,0 +1,214 @@
+//
+//  PreferencesWindowController.m
+//  PlutoHelperAgent
+//
+//  Created by Dave Allison on 26/02/2018.
+//  Copyright © 2018 Guardian News & Media. All rights reserved.
+//
+
+#import "PreferencesWindowController.h"
+
+@interface PreferencesWindowController ()
+
+@end
+
+@implementation PreferencesWindowController
+
+- (void)windowDidLoad {
+    [super windowDidLoad];
+    
+    UInt32 pwLength = 0;
+    
+    void* pwData = NULL;
+    
+    SecKeychainItemRef itemRef = NULL;
+    
+    NSString* service = @"PlutoHelperAgent";
+
+    OSStatus status = SecKeychainFindGenericPassword(
+                                                     
+                                                     NULL,         // Search default keychains
+                                                     
+                                                     (UInt32)service.length,
+                                                     
+                                                     [service UTF8String],
+                                                     
+                                                     0,
+                                                     
+                                                     NULL,
+                                                     
+                                                     &pwLength,
+                                                     
+                                                     &pwData,
+                                                     
+                                                     &itemRef      // Get a reference this time
+                                                     
+                                                     );
+    
+    if (status == errSecSuccess) {
+        
+        NSData* data = [NSData dataWithBytes:pwData length:pwLength];
+        
+        NSString* password = [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
+        
+        //NSLog(@"Read password %@", password);
+        
+        _PasswordText.stringValue = password;
+        
+        SecKeychainAttribute attrs2[] = {
+            
+            { kSecAccountItemAttr, 0, NULL }
+            
+        };
+        
+        SecKeychainAttributeList attributes2 = { 1, attrs2 };
+
+        OSStatus status5 = SecKeychainItemCopyContent(itemRef, NULL, &attributes2, NULL, NULL);
+        
+        if (status5 == errSecSuccess) {
+            
+            NSLog(@"Access sucessful");
```

Comment:
Please could you update these log lines to be more sepecific as to what has actually worked, e.g. "Password retrived from keystore".  If the software is not working and all we see in the Console log is "Access Successful", that does not really help much,

Reply:
Done.